### PR TITLE
Cypress/FE-2282 radio button

### PR DESCRIPTION
**Pull request by KatarzynaQA**
# Description

Added automation tests for RadioButton classic and default component

### CODE REVIEW

File: cypress/features/regression/radioButton/radioButtonWeeklyClassic.feature [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    Then "First" RadioButton component is enabled
+
+  @positive
+  Scenario: Enable reverse radioButton
+    When I check weekly reverse checkbox
+    Then "First" RadioButton is set to reverse
+
+  @positive
+  Scenario: Enable and disable reverse checkbox
+    # Given Given I open weekly tab
```
Comment (DlgSHi) [inline cypress/features/regression/radioButton/radioButtonWeeklyClassic.feature]: `Given`

File: cypress/features/regression/radioButton/radioButtonYearlyClassic.feature [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    Then "Third" RadioButton component is enabled
+
+  @positive
+  Scenario: Enable reverse radioButton
+    When I check yearly reverse checkbox
+    Then "Third" RadioButton is set to reverse
+
+  @positive
+  Scenario: Enable and disable reverse checkbox
+    # Given Given I open yearly tab
```
Comment (DlgSHi) [inline cypress/features/regression/radioButton/radioButtonYearlyClassic.feature]: `Given`

File: cypress/support/step-definitions/common-steps.js [hunk 3]
```diff
@@ -230,6 +265,21 @@ Then('labelAlign on preview is {string}', (direction) => {
   }
 });
 
+Then('{string} label Align on preview is {string}', (position, direction) => {
+  switch (position) {
+    case 'First':
+      labelByPosition(FIRST_ELEMENT).should('have.css', TEXT_ALIGN, `${direction}`);
```
Comment (DlgSHi) [inline cypress/support/step-definitions/common-steps.js]: no need to pass the parameter using ${parameter} if we are not combining strings, just pass `direction`

File: cypress/support/step-definitions/radio-button-steps.js [hunk 4]
```diff
@@ -17,14 +22,230 @@ Then('fieldHelpInline is disabled', () => {
...
+      break;
+    default: throw new Error('There are only three radio elements on the page');
+  }
+});
+
+Then('{string} RadioButton component is disabled', (position) => {
+  switch (position) {
+    case 'First':
+      radioButtonComponent(FIRST_RADIOBUTTON).should('have.attr', 'disabled');
+      radioButtonByPosition(FIRST_RADIOBUTTON).should('have.attr', 'disabled');
```
Comment (DlgSHi) [inline cypress/support/step-definitions/radio-button-steps.js]: we should add also `.should('be.disabled')` to check :)

File: cypress/support/step-definitions/radio-button-steps.js [hunk 5]
```diff
@@ -17,14 +22,230 @@ Then('fieldHelpInline is disabled', () => {
...
+      break;
+    default: throw new Error('There are only three radio elements on the page');
+  }
+});
+
+Then('{string} RadioButton component is enabled', (position) => {
+  switch (position) {
+    case 'First':
+      radioButtonComponent(SECOND_RADIOBUTTON).should('not.have.attr', 'disabled');
+      radioButtonByPosition(FIRST_RADIOBUTTON).should('not.have.attr', 'disabled');
```
Comment (DlgSHi) [inline cypress/support/step-definitions/radio-button-steps.js]: we should add also .should('not.be.disabled') to check :)

File: cypress/support/step-definitions/radio-button-steps.js [hunk 6]
```diff
@@ -17,14 +22,230 @@ Then('fieldHelpInline is disabled', () => {
...
+      radioButtonByPosition(SECOND_RADIOBUTTON).should('not.have.attr', 'disabled');
+      break;
+    case 'Third':
+      radioButtonComponent(SECOND_RADIOBUTTON).should('not.have.attr', 'disabled');
+      radioButtonByPosition(THIRD_RADIOBUTTON).should('not.have.attr', 'disabled');
+      break;
+    default: throw new Error('There are only three radio elements on the page');
+  }
+});
+// dokończyc
```
Comment (DlgSHi) [inline cypress/support/step-definitions/radio-button-steps.js]: unnecessary comment

File: cypress/support/step-definitions/radio-button-steps.js [hunk 7]
```diff
@@ -17,14 +22,230 @@ Then('fieldHelpInline is disabled', () => {
...
+        radioButtonByPosition(THIRD_RADIOBUTTON).should('have.css', 'width', '24px')
+          .and('have.css', 'height', '24px');
+        break;
+      default: throw new Error('There are only three radio elements on the page');
+    }
+  }
+});
+
+Then('{string} field help is set to fieldHelpInline', (position) => {
+  cy.wait(500, { log: DEBUG_FLAG });
```
Comment (DlgSHi) [inline cypress/support/step-definitions/radio-button-steps.js]: this wait was added due to ?

File: cypress/support/step-definitions/radio-button-steps.js [hunk 8]
```diff
@@ -17,14 +22,230 @@ Then('fieldHelpInline is disabled', () => {
...
+    case 'Third':
+      fieldHelpPreviewByPosition(THIRD_RADIOBUTTON).should('have.css', 'margin-left', '0px')
+        .and('have.css', 'margin-right', '6px');
+      break;
+    default: throw new Error('There are only three field help elements on the page');
+  }
+});
+
+Then('{string} field help is not set to fieldHelpInline and has margin-left set to {string}', (position, marginLeft) => {
+  cy.wait(500, { log: DEBUG_FLAG });
```
Comment (DlgSHi) [inline cypress/support/step-definitions/radio-button-steps.js]: also here about wait

File: cypress/features/regression/radioButton/radioButtonMonthly.feature [hunk 9]
```diff
@@ -0,0 +1,132 @@
+Feature: Experimental RadioButton monthly component
+  I want to change Experimental RadioButton component properties
+
+  Background: Open Experimental RadioButton component page
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/radioButton/radioButtonMonthly.feature]: fixed

File: cypress/features/regression/radioButton/radioButtonMonthly.feature [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+    Then "Second" RadioButton component is enabled
+
+  @positive
+  Scenario: Enable reverse radioButton
+    When I check monthly reverse checkbox
+    Then "Second" RadioButton is set to reverse
+
+  @positive
+  Scenario: Enable and disable reverse checkbox
+    # Given Given I open monthly tab
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/radioButton/radioButtonMonthly.feature]: fixed

File: cypress/features/regression/radioButton/radioButtonMonthlyClassic.feature [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+    Then "Second" RadioButton component is enabled
+
+  @positive
+  Scenario: Enable reverse radioButton
+    When I check monthly reverse checkbox
+    Then "Second" RadioButton is set to reverse
+
+  @positive
+  Scenario: Enable and disable reverse checkbox
+    # Given Given I open monthly tab
```
[the remark this answers is not in the document]
Comment (KatarzynaQA) [reply cypress/features/regression/radioButton/radioButtonMonthlyClassic.feature]: fixed

File: cypress/features/regression/radioButton/radioButtonWeeklyClassic.feature (hunk 1 above)
Comment (KatarzynaQA) [reply cypress/features/regression/radioButton/radioButtonWeeklyClassic.feature]: fixed

File: cypress/features/regression/radioButton/radioButtonYearlyClassic.feature (hunk 2 above)
Comment (KatarzynaQA) [reply cypress/features/regression/radioButton/radioButtonYearlyClassic.feature]: fixed

File: cypress/support/step-definitions/common-steps.js (hunk 3 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/common-steps.js]: fixed

File: cypress/support/step-definitions/radio-button-steps.js (hunk 6 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/radio-button-steps.js]: fixed

File: cypress/support/step-definitions/radio-button-steps.js (hunk 7 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/radio-button-steps.js]: this 'wait' is not necessary, deleted

File: cypress/support/step-definitions/radio-button-steps.js (hunk 8 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/radio-button-steps.js]: this 'wait' is not necessary, deleted

File: cypress/support/step-definitions/radio-button-steps.js (hunk 4 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/radio-button-steps.js]: fixed

File: cypress/support/step-definitions/radio-button-steps.js (hunk 5 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/radio-button-steps.js]: fixed

**Pull request merged**